web/src/Components/IssueCard.elm

```diff
@@ -22,119 +24,118 @@ issueCard model ballotId =
         ballot =
             getBallot ballotId model
 
+        vote =
+            getVoteFromBallot ballotId model
+
         ballotDone =
             ballot.finish < model.now
 
-        clickMsg =
-            if ballotDone then
-                Nav <| NTo <| ResultsR ballotId
-            else
-                Nav <| NTo <| VoteR ballotId
-
-        cardColor =
+        isPending =
+            ballot.state
+                == BallotPendingCreation
+                || ballot.state
+                == BallotPendingEdits
+                || ballot.state
+                == BallotPendingDeletion
+                || vote.state
+                == VotePending
+
+        state =
             if ballotDone then
-                vary (IssueCardMod IssuePast) True
+                IssuePast
             else if ballot.start > model.now then
-                vary (IssueCardMod VoteFuture) True
+                IssueFuture
+            else if isPending then
+                IssuePending
             else if checkAlreadyVoted ballotId model then
-                vary (IssueCardMod VoteDone) True
+                IssueDone
             else
-                vary (IssueCardMod VoteWaiting) True
-
-        resultString { name, result } =
-            name ++ " - " ++ toString (getResultPercent ballot <| result ? 0) ++ "%, "
-
-        --      [TS] Pretty messy, will probably need a refactor.
-        --      Also, Just picks the first result in a tie.
-        showWinningBallot =
-            case head (filter (\{ result } -> result == maximum (List.map (\{ result } -> result ? 0) ballot.ballotOptions)) ballot.ballotOptions) of
-                Just winningBallot ->
-                    winningBallot.name
-
-                Nothing ->
-                    "No Winner"
-
-        displayResults =
-            "Winning Result: " ++ showWinningBallot
-
-        voteStatus =
-            if ballotDone then
-                displayResults
-            else if ballot.start > model.now then
-                "This ballot is not open yet"
-            else if checkAlreadyVoted ballotId model then
-                "You have voted in this ballot"
-            else
-                "You have not voted in this ballot yet"
+                IssueVoteNow
+
+        struct =
```

Comment:
if any of the following 20 lines is common it should be generated programatically - like there's a LOT of similar code here. Maybe use a make function mapping over a list of tuples or something.

Reply:
This comment is just to make the above part of the review